### PR TITLE
code to review

### PR DESCRIPTION
**Pull request by kunalsingh3110**

### CODE REVIEW

File: CodeToReview.java [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+public class TicTacToe {
+
+    /**
+     * @param args the command line arguments
+     * @throws java.lang.InterruptedException
+     */
+    public static void main(String[] args) throws InterruptedException {
+        Scanner input = new Scanner(System.in);
+        String a1 = "1";
+        String a2 = "2";
```
Comment (kunalsingh3110) [inline CodeToReview.java:28]: @beast-incarnate, I believe here you can easily use an array instead of creating 9 strings.

File: CodeToReview.java [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+        int inO;
+        int inX = 0;
+        boolean winner = false;
+        PrintBoard(a1,a2,a3,a4,a5,a6,a7,a8,a9);
+        System.out.printf("How to play - wait for your turn and when it comes write the number of the field that you want to draw in (from 1 to 9)\n");
+        System.out.printf("Player O's turn: ");
+        inO = input.nextInt();
+        while (true) {
+            switch (inO) {
+                case 1:
```
Comment (kunalsingh3110) [inline CodeToReview.java:45]: @beast-incarnate, this seems to be an infinite loop to me. Please recheck this and also try to avoid multiple
 switch branches. It will lead to more problems than fixes.

File: CodeToReview.java [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+                    }
+                    break;
+                default:
+                    System.out.printf("An error occured. Please accept it politely and restart the game.");
+                    break;
+            }
+            System.out.printf("\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n");
+            if (null != Winners(a1,a2,a3,a4,a5,a6,a7,a8,a9))switch (Winners(a1,a2,a3,a4,a5,a6,a7,a8,a9)) {
+                case "NONE":
+                    if (a1 != "1" && a2 != "2" && a3 != "3" && a4 != "4" && a5 != "5" && a6 != "6" && a7 != "7" && a8 != "8" && a9 != "9") {
```
Comment (kunalsingh3110) [inline CodeToReview.java:97]: @beast-incarnate, try avoiding such a long condition in an if. It's not readable and will be hard in future to test edge cases.

File: CodeToReview.java [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+        System.out.printf("   %s | %s | %s \n",f1,f2,f3);
+        System.out.printf("  ----------- \n");
+        System.out.printf("   %s | %s | %s \n",f4,f5,f6);
+        System.out.printf("  ----------- \n");
+        System.out.printf("   %s | %s | %s \n",f7,f8,f9);
+        System.out.printf(" \n");
+    }
+    
+    public static String Winners(String f1, String f2, String f3, String f4, String f5, String f6, String f7, String f8, String f9) {
+        if (f1 == f2 && f2 == f3 && f3 == "O") {
```
Comment (kunalsingh3110) [inline CodeToReview.java:217]: @beast-incarnate, try limiting comparison in an if to a minimum or moving them to a function. And I believe we can achieve the required functionality without using such a long if-else ladder.

File: CodeToReview.java [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+                case 9:
+                    if (a9 == "9") {
+                    a9 = "X";
+                    }
+                    break;
+                default:
+                    System.out.printf("An error occured. Please accept it politely and restart the game.");
+                    break;
+            }
+            System.out.printf("\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n");
```
Comment (kunalsingh3110) [inline CodeToReview.java:174]: @beast-incarnate , avoid such unnecessary prints

File: CodeToReview.java [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+                    System.out.printf("O HAS WON!!!");
+                    Thread.sleep(3000);
+                    break;
+                default:
+                    break;
+            }
+            if (winner) {
+                break;
+            }
+            switch (inX) {
```
Comment (kunalsingh3110) [inline CodeToReview.java:124]: @beast-incarnate , I believe this functionality can be achieved from a function rather than using such a large switch case block.

File: CodeToReview.java [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+                    }
+                    break;
+                default:
+                    System.out.printf("An error occured. Please accept it politely and restart the game.");
+                    break;
+            }
+            System.out.printf("\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n");
+            if (null != Winners(a1,a2,a3,a4,a5,a6,a7,a8,a9))switch (Winners(a1,a2,a3,a4,a5,a6,a7,a8,a9)) {
+                case "NONE":
+                    if (a1 != "1" && a2 != "2" && a3 != "3" && a4 != "4" && a5 != "5" && a6 != "6" && a7 != "7" && a8 != "8" && a9 != "9") {
```
Comment (kunalsingh3110) [inline CodeToReview.java:177]: @beast-incarnate , very long if condition. we should try to keep it at a minimum or create a function for this.
